packages/instrumentation-openai/test/fixtures.test.js

```diff
@@ -0,0 +1,1354 @@
+if (process.env.TEST_FIXTURES_ENV_FILE) {
+  require('dotenv').config({ path: process.env.TEST_FIXTURES_ENV_FILE });
+}
+
+const UNIT_TEST_MODEL_TOOLS = 'gpt-4o-mini';
+const UNIT_TEST_MODEL_EMBEDDINGS = 'text-embedding-3-small';
+
+// Configure the test fixtures based on the test mode.
+const testMode = process.env.TEST_FIXTURES_MODE || 'unit';
+const isUnit = testMode === 'unit'; // shorthand for terser code below
+// Nock is used for HTTP-mocking in some test modes. It interferes with
+// `@opentelemetry/instrumentation-http`, so we skip assertions for HTTP
+// spans when using nock.
+let usingNock = false;
+let targetService;
+
+switch (testMode) {
+  case 'unit':
+    // Unit tests.
+    // "unit" mode is for running using pre-recorded OpenAI API responses.
+    // This is implemented using `nock`s "Nock back" feature.
+    usingNock = true;
+    // https://github.com/nock/nock#modes
+    process.env.TEST_NOCK_BACK_MODE = 'lockdown';
+    // OPENAI_API_KEY needs to be set to something to avoid OpenAI
+    // constructor error. However, because of mocking, it isn't used.
+    process.env.OPENAI_API_KEY = 'notused';
+    process.env.TEST_MODEL_TOOLS = UNIT_TEST_MODEL_TOOLS;
+    process.env.TEST_MODEL_EMBEDDINGS = UNIT_TEST_MODEL_EMBEDDINGS;
+    targetService = 'openai';
+    break;
+
+  case 'regenerate-recordings':
+    // Regenerate the recorded OpenAI API responses use by "unit" test mode.
+    if (process.env.OPENAI_BASE_URL) {
+      throw new Error(
+        'OPENAI_BASE_URL is set. To regenerate-recordings, it must be empty so that recorded responses are from api.openai.com'
+      );
+    }
+    if (!process.env.OPENAI_API_KEY) {
+      throw new Error(
+        'OPENAI_API_KEY is not set. To regenerate-recordings, it must be set. Set it in your environment, or use TEST_FIXTURES_ENV_FILE="./openai.env" (see "openai.env.template").'
+      );
+    }
+    usingNock = true;
+    process.env.TEST_NOCK_BACK_MODE = 'update';
+    process.env.TEST_MODEL_TOOLS = UNIT_TEST_MODEL_TOOLS;
+    process.env.TEST_MODEL_EMBEDDINGS = UNIT_TEST_MODEL_EMBEDDINGS;
+    targetService = 'openai';
+    break;
+
+  case 'integration':
+    // Guess the target service based on the config in the environment.
+    // This is necessary because there some small differences between Azure
+    // OpenAI, OpenAI, and Ollama that test assertions need to cope with.
+    if (
+      process.env.OPENAI_BASE_URL &&
+      new URL(process.env.OPENAI_BASE_URL).port === '11434'
+    ) {
+      targetService = 'ollama';
+    } else if (process.env.AZURE_OPENAI_API_KEY) {
+      targetService = 'azure';
+    } else {
+      targetService = 'openai';
+    }
+    break;
+
+  default:
+    throw new Error('unknown test mode: ' + testMode);
+}
+
+// ---- helper functions
+
+function isPositiveInteger(val) {
+  return Number.isInteger(val) && val > 0;
+}
+
+function isExpectedServerAddress(val) {
```

Comment:
unless targetService is used in other ways, for address and port, could it be generalized to one of AZURE_OPENAI_ENDPOINT,  OPENAI_BASE_URL, or default url, then parse that the same way regardless of assumed service name?

Reply:
commit 423fb34